odoo/models.py

```diff
@@ -1813,35 +1813,20 @@ def _read_group_fill_temporal(self, data, groupby, aggregated_fields, annotated_
 
         expected = collections.deque(date_utils.date_range(first, last, interval))
 
-        if len(existing) < len(expected):
-            empty_data = dict.fromkeys(aggregated_fields, False)
-            empty_data['id'] = False
-            empty_data[groupby_name.split(':')[0] + '_count'] = 0
-
-            data = collections.deque(data)
-
-            new_data = []
-
-            # Note: the list 'expected' contains the the dates that should be
-            # represented inside data when it is returned. Notice that 'expected'
-            # is sorted like data, so what we do is empty data progressivly by
-            # popping its elements from first to last. To do this properly, we
-            # need to compare dates together.
-            while expected or data:
-                if not data[0][groupby_name]:
-                    new_data.append(data.popleft())
-                    continue
-                dt = expected.popleft()
-                if data[0][groupby_name] == dt:
-                    new_data.append(data.popleft())
-                else:
-                    new_data.append(dict(empty_data, **{groupby_name: dt}))
-
-            assert not data
-            return new_data
-
-        return data
-
+        new_data = collections.defaultdict(list)
+        for d in data:
+            new_data[d[groupby_name]].append(d)
```

Comment:
The name `new_data` is confusing, because not the same data structure as `data`.
I suggest `grouped_data`.